plugin.js

```diff
@@ -106,18 +115,24 @@ function plugin (fastify, options, next) {
   }
 
   fastify.decorateRequest('cookies', null)
-  fastify.decorateReply('cookie', setCookie)
+  fastify.decorateReply(kReplySetCookies, null)
 
+  fastify.decorateReply('cookie', setCookie)
   fastify.decorateReply('setCookie', setCookie)
   fastify.decorateReply('clearCookie', clearCookie)
 
   if (hook) {
     fastify.addHook(hook, onReqHandlerWrapper(fastify, hook))
+    fastify.addHook('onSend', onSendHandlerWrapper(fastify))
```

Comment:
If you accept above change, than you have to do:

```suggestion
    fastify.addHook('onSend', fastifyCookieOnSendHandler)
```